hepstats/hypotests/calculators/basecalculator.py

```diff
@@ -436,3 +303,176 @@ def q(self, nll1: np.array, nll2: np.array, poi1, poi2,
+                sampler : function used to create sampler with models, number of events and
+                    floating parameters in the sample Default is `hepstats.fitutils.sampling.base_sampler`.
+                sample : function used to get samples from the sampler.
+                    Default is `hepstats.fitutils.sampling.base_sample`.
+        """
+        super(BaseToysCalculator, self).__init__(input, minimizer)
+
+
+class ToysCalculator(BaseToysCalculator, ToysManager):
+    """
+    Class for calculators using toys.
+    """
+
+    def __init__(self, input, minimizer, ntoysnull=100, ntoysalt=100, sampler=base_sampler, sample=base_sample):
+        """Toys calculator class.
+
+            Args:
+                input : loss or fit result
+                minimizer : minimizer to use to find the minimum of the loss function
+                ntoysnull (int, default=100): minimum number of toys to generate for the null hypothesis
+                ntoysalt (int, default=100): minimum number of toys to generate for the alternative hypothesis
+                sampler : function used to create sampler with models, number of events and
+                    floating parameters in the sample Default is `hepstats.fitutils.sampling.base_sampler`.
+                sample : function used to get samples from the sampler.
+                    Default is `hepstats.fitutils.sampling.base_sample`.
+        """
+        super(ToysCalculator, self).__init__(input, minimizer, sampler, sample)
+
+        self._ntoysnull = ntoysnull
+        self._ntoysalt = ntoysalt
+
+    @classmethod
+    def from_yaml(cls, filename, loss, minimizer, ntoysnull=100, ntoysalt=100, sampler=base_sampler,
+                  sample=base_sample):
+        """
+        ToysCalculator constructor with the toys loaded from a yaml file.
+
+        Args:
+            filename (str)
+            input : loss or fit result
+            minimizer : minimizer to use to find the minimum of the loss function
+            ntoysnull (int, default=100): minimum number of toys to generate for the null hypothesis
+            ntoysalt (int, default=100): minimum number of toys to generate for the alternative hypothesis
+            sampler : function used to create sampler with models, number of events and
+                floating parameters in the sample Default is `hepstats.fitutils.sampling.base_sampler`.
+            sample : function used to get samples from the sampler.
+                Default is `hepstats.fitutils.sampling.base_sample`.
+
+        Returns
+            ToysCalculator
+        """
+
+        calculator = cls(loss, minimizer, ntoysnull, ntoysalt, sampler, sample)
+        toysresults = calculator.toysresults_from_yaml(filename)
+
+        for t in toysresults:
+            calculator.add_toyresult(t)
+
+        return calculator
+
+    @property
+    def ntoysnull(self):
+        """
+        Returns the number of toys loss for the null hypothesis.
```

Comment:
I do not understand the sentence - "toys loss"? Same for similar property below.

Reply:
I will replace `loss` by generated.

Reply:
Now I do understand :-).